Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -6,7 +6,9 @@ def fix_dynamic_linkage
 
         each_install_name_for(file) do |bad_name|
           # Don't fix absolute paths unless they are rooted in the build directory
-          next if bad_name.start_with?("/") && !bad_name.start_with?(HOMEBREW_TEMP.to_s)
+          next if bad_name.start_with?("/") &&
+                  !bad_name.start_with?(HOMEBREW_TEMP.to_s) &&
+                  !bad_name.start_with?(HOMEBREW_TEMP.realpath.to_s)
```

Comment:
Ah good to know. So `readlink` requires a symlink but can read a bad link, and `realpath` will accept non-symlinks, but dies on a bad link. Sometimes I wonder if these functions are out to get us.